### PR TITLE
Added method catIf()

### LINKED ISSUE #3: New Method: catIf(arg1, arg2,..., argN, flag)
It’s very common that you will need perform the string concatenation only if a constraint is satisfied.  The last argument of this method must be a boolean testable value that will indicate if the given strings shall be added or not.
```js
var sb = new StringBuilder();
sb.cat(‘Hello’)
    .catIf(‘ pretty’, ‘ lady!’, sex === ‘f’)
    .catIf(‘ gentleman!’, sex === ‘m’)
    .catIf(‘ and’, ‘ good’, ‘bye!’, !sex);
```
As you can see this method is very useful if you are writing an application that discriminates people that have sex undefined. :-)

### PR DESCRIPTION
**Pull request by islas27**
- Added documentation
- Added unit test

Branch based on feature/method-cat 5ea3ee1 and closes #3

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  }
+
+  this.str = function () {
+    return buffer.join('')
+  }
+
+  this.catIf = function () {
+    let args = [...arguments]
+    const idx = args.length - 1
+    if (idx > 0 && classifier(args[idx]) === 'boolean' && args[idx]) {
```
Comment (ryanez) [inline index.js]: What about having a method like `classifier` that allows us to test for booleans? like `isBool(args[idx])`

File: index.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    }
+    return this
+  }
+
+  this.str = function () {
+    return buffer.join('')
+  }
+
+  this.catIf = function () {
+    let args = [...arguments]
```
Comment (ryanez) [inline index.js:30]: You can always `pop` the array.

Comment (islas27) [reply index.js:30]: rewrote the function, now it looks way nicer: fc2eb78
what do you think? @ryanez

File: index.js [hunk 3]
```diff
@@ -23,5 +24,12 @@ function stringHelper () {
     return buffer.join('')
   }
 
+  this.catIf = function () {
+    let args = [...arguments]
+    let cond = args.pop()
+    if (isType('boolean', cond) && cond) this.cat(args)
```
Comment (ryanez) [inline index.js]: would the `cond === true` do the same trick?

Comment (islas27) [reply index.js]: well, I reduced it even more, so it became: 
```js
if (args.pop() === true) this.cat(args)
```
Commit e55cc45

File: README.md [hunk 4]
```diff
@@ -58,3 +58,22 @@ This function will 'save' all the strings sent to it in any format, be it a list
 **Notes**:
 - When `typeof` of an input evaluates to `object` or `undefined`, this argument will be ignored. This happens to any kind of object, map, null or undefined values. To know more about the `typeof` operand, follow this link: [typeof - MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof)
 - While arrow functions may be more clean and concise, they do not bind a context to the execution of the function (`this`), so in case you need to use the helper inside a function that is going to be passed as an argument, write an old fashion function. More info on arrow functions: [Arrow Functions - MDN](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Functions/Arrow_functions)
+
+### catIf(..args, condition)
+This function will 'save' all the strings sent to it in any format, be it a list of strings, an array of strings, a function even can be sent and it will be executed, saving the result for further processes.
```
Comment (ryanez) [inline README.md]: What you mean with -> *a function even can be sent and it will be execute, saving the result for further processes*?

Comment (islas27) [reply README.md]: Improved the documentation (method explanations) in 70d6d72

File: README.md [hunk 5]
```diff
@@ -57,7 +57,28 @@ This function will 'save' all the strings sent to it in any format, be it a list
...
+- While arrow functions may be more clean and concise, they do not bind a context to the execution of the function (`this`), so in case you need to use the helper inside a function that is going to be passed as an argument, write an old fashioned function. More info on arrow functions: [Arrow Functions - MDN](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Functions/Arrow_functions)
+
+---
+
+### catIf(..args, condition)
+This function works as a conditional wrapper for `cat()`, as it will only execute `cat()` if the condition sent evaluates to true.
+
+**Sintax**: `helper.catIf(arg1, arg2, ...argN, contidion)`
+
+**Inputs**: `argN`: Any kind of input that ultimately can be converted or executed into a string. `contidion`: a boolean (strictly) to decide is the previous arguments are to be cat() or not.
```
Comment (ryanez) [inline README.md]: to decide "if"

**Review by ryanez — COMMENTED** [review]
everything else LGTM

**Pull request merged**